caikit_nlp/modules/text_embedding/embedding.py

```diff
@@ -0,0 +1,142 @@
+)
+class EmbeddingModule(ModuleBase):
+
+    _ARTIFACTS_PATH_KEY = "artifacts_path"
+    _ARTIFACTS_PATH_DEFAULT = "artifacts"
+
+    def __init__(
+        self,
+        model: SentenceTransformer,
+    ):
+        super().__init__()
+        self.model = model
+
+    @classmethod
+    def load(cls, model_path: str, *args, **kwargs) -> "EmbeddingModule":
+        """Load model
+
+        Args:
+            model_path: str
+                Path to the config dir under the model_id (where the config.yml lives)
+
+        Returns:
+            EmbeddingModule
+                Instance of this class built from the model.
+        """
+
+        config = ModuleConfig.load(model_path)
+        artifacts_path = config.get(cls._ARTIFACTS_PATH_KEY)
+
+        error.value_check(
+            "<NLP07391618E>",
+            artifacts_path,
+            ValueError(f"Model config missing '{cls._ARTIFACTS_PATH_KEY}'"),
+        )
+
+        artifacts_path = os.path.abspath(os.path.join(model_path, artifacts_path))
+        error.dir_check("<NLP34197772E>", artifacts_path)
+
+        return cls.bootstrap(model_name_or_path=artifacts_path)
+
+    def run(
+        self, input: str, **kwargs  # pylint: disable=redefined-builtin
+    ) -> EmbeddingResult:
+        """Run inference on model.
+        Args:
+            input: str
+                Input text to be processed
+        Returns:
+            EmbeddingResult: the result vector nicely wrapped up
+        """
+        error.type_check("<NLP27491611E>", str, input=input)
+
+        return EmbeddingResult(Vector1D.from_vector(self.model.encode(input)))
+
+    @classmethod
+    def bootstrap(cls, model_name_or_path: str) -> "EmbeddingModule":
+        """Bootstrap a sentence-transformers model
+
+        Args:
+            model_name_or_path: str
+                Model name (Hugging Face hub) or path to model to load.
+        """
+        return cls(model=SentenceTransformer(model_name_or_path=model_name_or_path))
+
+    def save(self, model_path: str, *args, **kwargs):
+        """Save model using config in model_path
+
+        Args:
+            model_path: str
+                Path to model config
+        """
+
+        model_config_path = model_path  # because the param name is misleading
+
+        error.type_check("<NLP82314992E>", str, model_path=model_config_path)
+        error.value_check(
+            "<NLP40145207E>",
+            model_config_path is not None and model_config_path.strip(),
+            f"model_path '{model_config_path}' is invalid",
+        )
+
+        model_config_path = os.path.abspath(
+            model_config_path.strip()
+        )  # No leading/trailing spaces sneaky weirdness
+
+        os.makedirs(model_config_path, exist_ok=False)
+        saver = ModuleSaver(
+            module=self,
+            model_path=model_config_path,
+        )
+
+        # Get and update config (artifacts_path)
+        artifacts_path = saver.config.get(self._ARTIFACTS_PATH_KEY)
```

Comment:
this will always be none right? Unless we are overriding the model ?

Reply:
right, I can remove this